src/main/java/edu/virginia/lib/indexing/ASpaceObject.java

```diff
@@ -594,8 +594,11 @@ else if (title.startsWith("The "))
                 barcode = topContainer.record.getString("barcode");
             } else {
                 String uri = topContainer.record.getString("uri");
-                String topContainerNumber = uri.substring(uri.lastIndexOf("/") + 1);
-                String repository = uri.split("/")[2];
+                Pattern pattern = Pattern.compile("/repositories/(\\d+)/(\\w+)/(\\d+)");
+                Matcher matcher = pattern.matcher(uri);
+                matcher.find();
+                String repository = matcher.group(1);
```

Comment:
If matcher.find() returns false, this line will throw an exception.  

Since we'd probably just throw an exception anyway, you need not update the code, but I'd typically use the form: if (matcher.find()) { // your code here } else { throw new RuntimeException(uri + " was of an unexpected format!"); }